src/prefect/agent/local/agent.py

```diff
@@ -65,6 +65,7 @@ def deploy_flows(self, flow_runs: list) -> None:
             env_vars = self.populate_env_vars(flow_run=flow_run)
 
             if not self.no_pull:
+                self.logger.debug("Pulling image {}...".format(storage.name))
```

Comment:
So this will be logged every time even if the image already exists locally. If you're cool with that we can leave it

Reply:
Yea we can change the wording if you'd like -> I just remember both of us wishing we saw a log like this during one of our onboards.

Reply:
I'm cool with it for now 👍

Reply:
Actually, what if we make it `Pulling image {} if it does not already exist...` or something along those lines

Reply:
I think it's still _technically_ a pull, it just happens to use a local cache.